lib/security/key_helper.dart

```diff
@@ -0,0 +1,252 @@
+    final signer = pc.RSASigner(pc.SHA256Digest(), '0609608648016503040201');
+
+    signer.init(true, pc.PrivateKeyParameter<pc.RSAPrivateKey>(privateKey));
+
+    final sig = signer.generateSignature(Uint8List.fromList(
+      utf8.encode(dataToSign),
+    ));
+
+    return sig.bytes;
+  }
+
+  bool rsaVerify(String publicKeyPem, String signedData, Uint8List signature) {
+    var publicKey = decodePublicKeyFromPem(publicKeyPem);
+
+    final sig = pc.RSASignature(signature);
+
+    final verifier = pc.RSASigner(pc.SHA256Digest(), '0609608648016503040201');
+
+    verifier.init(
+      false,
+      pc.PublicKeyParameter<pc.RSAPublicKey>(publicKey),
+    ); // false=verify
+
+    try {
+      return verifier.verifySignature(
+        Uint8List.fromList(signedData.codeUnits),
+        sig,
+      );
+    } on ArgumentError {
+      return false;
+    }
+  }
+
+  pc.SecureRandom _generateeSecureRandom() {
+    final secureRandom = pc.FortunaRandom();
+
+    final seedSource = Random.secure();
+    final seeds = <int>[];
+    for (var i = 0; i < 32; i++) {
+      seeds.add(
+        seedSource.nextInt(255),
+      );
+    }
+    secureRandom.seed(
+      pc.KeyParameter(
+        Uint8List.fromList(seeds),
+      ),
+    );
+
+    return secureRandom;
+  }
+
+  pc.RSAPrivateKey decodePrivateKeyFromPem(String priavateKeyPem) {
+    try {
+      var key = encrypt_pem.RSAKeyParser()
+          .parse(priavateKeyPem.replaceAll('\\r\\n', '\n')) as pc.RSAPrivateKey;
+
+      return key;
+    } on FormatException {
+      return encrypt_pem.RSAKeyParser().parse(priavateKeyPem)
+          as pc.RSAPrivateKey;
+    }
+  }
+
+  pc.RSAPublicKey decodePublicKeyFromPem(String publicKeyPem) {
+    try {
+      var key = encrypt_pem.RSAKeyParser()
+          .parse(publicKeyPem.replaceAll('\\r\\n', '\n')) as pc.RSAPublicKey;
+
+      return key;
+    } on FormatException {
+      try {
+        var key =
+            encrypt_pem.RSAKeyParser().parse(publicKeyPem) as pc.RSAPublicKey;
+        ;
+
+        return key;
+      } on FormatException {
+        return encrypt_pem.RSAKeyParser().parse(publicKeyPem)
+            as pc.RSAPublicKey;
+      }
+    }
+  }
+
+  String encryptWithPublicKeyPem(String publicKeyPem, String message) {
+    var publicKey = decodePublicKeyFromPem(publicKeyPem);
+
+    return encrypt(publicKey, message);
+  }
+
+  String decryptWithPrivateKeyPem(String privateKeyPem, String message) {
+    var privateKey = decodePrivateKeyFromPem(privateKeyPem);
+
+    return decrypt(privateKey, message);
+  }
+
+  String encodePublicKeyToPem(pc.RSAPublicKey publicKey) {
+    var topLevel = ASN1Sequence();
+
+    topLevel.add(
+      ASN1Integer(publicKey.modulus!),
+    );
+    topLevel.add(
+      ASN1Integer(publicKey.exponent!),
+    );
+
+    var dataBase64 = base64.encode(topLevel.encodedBytes);
+
+    return '-----BEGIN RSA PUBLIC KEY-----\r\n$dataBase64\r\n-----END RSA PUBLIC KEY-----';
```

Comment:
Also maybe add a function to wrap the key in the 'BEGIN ... END' since it is done more than once and largely contains string constants.